soc/hps_soc.py

```diff
@@ -248,7 +249,9 @@ def main():
         integrated_rom_init = []
 
     if args.cpu_cfu:
-        if args.slim_cpu:
+        if args.cpu_variant:
+            variant = args.cpu_variant
+        elif args.slim_cpu:
```

Comment:
Why is there both slim_cpu and cpu_variant? Is this supposed to be temporary? If so, perhaps print a message when the old form is used, and also note in the help string for the slim_cpu option.

Reply:
Yes, you're right, there's overlapping capability, the old specific version (--slim_cpu) and the new general version (--cpu-variant=slim+cfu).    I should deprecate the old version, and also update any uses of the old version in project Makefiles.